libs/barista-components/stacked-series-chart/src/stacked-series-chart.html

```diff
@@ -31,11 +31,17 @@
     </span>
 
     <div
-      class="dt-stacked-series-chart-track"
+      class="dt-stacked-series-chart-track dt-stacked-series-chart-{{
+        selectionMode
+      }}"
       [class.dt-stacked-series-chart-track-selectable]="_selectable"
+      [class.dt-stacked-series-chart-track-selected]="track.selected"
       [class.dt-stacked-series-chart-track-background]="visibleTrackBackground"
       [class.dt-stacked-series-chart-track-hoverable]="_overlay"
       [style]="_sanitizeCSS([['--dt-stacked-series-chart-track-index', i + 1]])"
+      (click)="
+        selectionMode === 'stack' && _toggleSelect(track.origin, undefined)
+      "
```

Comment:
Please perform the check inside the `_toggleSelect` function. try to keep the templates as simple as possible simple if in the beginning then you can check fro stack and node.

Reply:
this looked clean to me, I created two functions to check that condition before calling the _toggleSelect and removed this logic from the template